tests/test_basic_s3.py

```diff
@@ -203,6 +205,29 @@ async def test_can_get_and_put_object(s3_client, create_object, bucket_name):
     assert data == b'body contents'
 
 
+@pytest.mark.moto
+@pytest.mark.asyncio
+@pytest.mark.config_kwargs(dict(retries={"max_attempts": 5, "mode": "adaptive"}))
+async def test_adaptive_retry(s3_client, create_object, bucket_name):
+    called = []
+
+    def _wrapt_failed(wrapped, instance, args, kwargs):
+        called.append(wrapped.__name__)
+        return wrapped(*args, **kwargs)
+
+    # Check that we did not call any blocking method
+    # Unfortunately can't patch threading.Lock.__enter__
+    with wrapt_function(
+        'botocore.retries.adaptive',
+        'ClientRateLimiter.on_sending_request',
+        _wrapt_failed
+    ), wrapt_function('botocore.retries.bucket', 'TokenBucket.acquire', _wrapt_failed):
+        await create_object('foobarbaz', body='body contents')
+        _ = await s3_client.get_object(Bucket=bucket_name, Key='foobarbaz')
```

Comment:
we should have a positive test here as well to ensure the new methods are called

Reply:
can be w/in this test, just wrap the new methods and ensure they're in called